examples/efficient_net.py

```diff
@@ -0,0 +1,298 @@
+        x = img_input
+        # x = layers.ZeroPadding3D(padding=self.correct_pad(x, 3),
+        #                         name='stem_conv_pad')(x)
+
+        x = layers.Conv3D(self.round_filters(32), 3,
+                        strides=2,
+                        padding='same',
+                        use_bias=False,
+                        kernel_initializer=CONV_KERNEL_INITIALIZER,
+                        name='stem_conv')(x)
+        x = layers.BatchNormalization(axis=bn_axis, name='stem_bn')(x)
+        x = layers.Activation(self.activation_fn, name='stem_activation')(x)
+        blocks_args = deepcopy(DEFAULT_BLOCKS_ARGS)
+
+        b = 0
+        # 计算总的block的数量
+        blocks = float(sum(args['repeats'] for args in blocks_args))
+        for (i, args) in enumerate(blocks_args):
+            assert args['repeats'] > 0
+            args['filters_in'] = self.round_filters(args['filters_in'])
+            args['filters_out'] = self.round_filters(args['filters_out'])
+
+            for j in range(self.round_repeats(args.pop('repeats'))):
+                if j > 0:
+                    args['strides'] = 1
+                    args['filters_in'] = args['filters_out']
+                x = self.block(x, self.activation_fn, self.drop_connect_rate * b / blocks,
+                        name='block{}{}_'.format(i + 1, chr(j + 97)), **args)
+                b += 1
+        
+        x = layers.Conv3D(self.round_filters(1280), 1,
+                        padding='same',
+                        use_bias=False,
+                        kernel_initializer=CONV_KERNEL_INITIALIZER,
+                        name='top_conv')(x)
+        x = layers.BatchNormalization(axis=bn_axis, name='top_bn')(x)
+        x = layers.Activation(self.activation_fn, name='top_activation')(x)
+
+        # Use GlobalAveragePooling2D replace Fully-connected layer
+        # x = layers.GlobalAveragePooling3D(name='avg_pool')(x)
+        print("input.shape", inputs.shape, x.shape)
+        # if dropout_rate > 0:
+        #     x = layers.Dropout(dropout_rate, name='top_dropout')(x)
+
+        # x = layers.Dense(classes,
+        #                     activation='softmax',
+        #                     kernel_initializer=DENSE_KERNEL_INITIALIZER,
+        #                     name='probs')(x)
+
+        return x
+
+    # 保证filter的大小可以被8整除
+    def round_filters(self, filters):
+        """Round number of filters based on depth multiplier."""
+        filters *= self.width_coefficient
+        divisor = self.depth_divisor
+        new_filters = max(divisor, int(filters + divisor / 2) // divisor * divisor)
+        # Make sure that round down does not go down by more than 10%.
+        if new_filters < 0.9 * filters:
+            new_filters += divisor
+        return int(new_filters)
+
+    # 重复次数，取顶
+    def round_repeats(self, repeats):
+        return int(math.ceil(self.depth_coefficient * repeats))
+
+
+config_path = "examples/config_efficient_net.yaml"
```

Comment:
please define `if __nam__ == "__main__":` to wrap things, so that when people import classes from this file, they will not execute the training